src/headers/header_name.rs

```diff
@@ -71,6 +71,30 @@ impl<'a> std::convert::TryFrom<&'a str> for HeaderName {
     }
 }
 
+impl PartialEq<str> for HeaderName {
+    fn eq(&self, other: &str) -> bool {
+        self.0 == other
```

Comment:
This is incorrect, header name comparisons must be case-insensitive.

Reply:
Oh yeah, good catch. We already normalize when casting from a string to a headername. But we should be doing the same for this comparison.